src/SystemServicesContext.tsx

```diff
@@ -0,0 +1,135 @@
+import React, {
+  createContext,
+  useState,
+  useEffect,
+  FunctionComponent,
+  useContext,
+} from "react"
+import { AppState, Platform } from "react-native"
+
+import { isBluetoothEnabled } from "./gaen/nativeModule"
+import { isLocationEnabled } from "./gaen/nativeModule"
+import { doesDeviceSupportLocationlessScanning } from "./gaen/nativeModule"
+
+export interface SystemServicesState {
+  isBluetoothOn: boolean
+  isLocationOn: boolean
+  isLocationNeeded: boolean
+}
+
+const initialState: SystemServicesState = {
+  isBluetoothOn: false,
+  isLocationOn: false,
+  isLocationNeeded: true,
+}
+
+export const SystemServicesContext = createContext<SystemServicesState>(
+  initialState,
+)
+
+const SystemServicesProvider: FunctionComponent = ({ children }) => {
+  const [isBluetoothOn, setIsBluetoothOn] = useState(false)
+  const [isLocationOn, setIsLocationOn] = useState(false)
+  const [isLocationNeeded, setIsLocationNeeded] = useState(false)
+
+  const fetchBTStatus = async (): Promise<boolean> => {
+    return isBluetoothEnabled()
+  }
+
+  useEffect(() => {
+    const determineIsBluetoothOn = async () => {
+      const status = await fetchBTStatus()
+      setIsBluetoothOn(status)
+    }
+
+    determineIsBluetoothOn()
+
+    const handleAppStateChange = () => {
+      determineIsBluetoothOn()
+    }
+    addListener(handleAppStateChange)
+    return removeListener(handleAppStateChange)
+  }, [])
+
+  const fetchIsLocationOn = async (): Promise<boolean> => {
+    return Platform.select({
+      android: isLocationEnabled(),
+      ios: new Promise(() => true),
```

Comment:
Yeah this strikes me as a bit odd.  We are saying location is on, but we in fact do not know one way or the other.

Not sure its worth updating at this point, but I would lean towards trying to encode what we know about the world more directly

Reply:
My thinking here is that it doesn't location is, by default, always enabled on iOS, and it doesn't really add anything to encode the fact that we don't know. It also may make working with this value on the view layer more challenging because we'd have to check if this is false, true, or "unknown", which will behave the same as it being true. Is there something off with my thinking there?

Reply:
I'll take a look at how this would change if we encode this as unknown and circle back. Curious…

Reply:
Well, turns out we do know whether location is on on iOS, so we can just use `isLocationEnabled()` directly. I won't change any behavior, but we'll have the actual value.